### ISSUE TITLE
BUG: Interchange with Pyarrow types loses validity buffer

### ISSUE THREAD

**Issue by WillAyd** [pandas-dev/pandas#56805]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
>>> df = pd.DataFrame([[None]], dtype="int32[pyarrow]")
>>> df.__dataframe__().get_column(0).describe_null
(<ColumnNullType.NON_NULLABLE: 0>, None)
>>> df.__dataframe__().get_column(0).get_buffers()
{'data': (PandasBuffer({'bufsize': 8, 'ptr': 100388886100432, 'device': 'CPU'}), (<DtypeKind.INT: 0>, 32, 'i', '=')), 'validity': None, 'offsets': None}
```

### Issue Description

No mention of nullability even though it is

### Expected Behavior

We should at least be sending a bit or a byte mask along through the dataframe api

### Installed Versions

main

**Comment by MarcoGorelli** [comment]
yeah this wrong - the Polars implementation looks correct here
```python
In [6]: pl.DataFrame({'a': [None]}, schema={'a': pl.Int32}).__dataframe__().get_column(0).get_buffers()
Out[6]:
{'data': (PolarsBuffer(bufsize=4, ptr=139918198210568, device='CPU'),
  (<DtypeKind.INT: 0>, 32, 'i', '=')),
 'validity': (PolarsBuffer(bufsize=1, ptr=139918198210576, device='CPU'),
  (<DtypeKind.BOOL: 20>, 1, 'b', '=')),
 'offsets': None}

In [7]: pd.DataFrame({'a': [None]}, dtype='int32[pyarrow]').__dataframe__().get_column(0).get_buffers()
Out[7]:
{'data': (PandasBuffer({'bufsize': 8, 'ptr': 93950600639216, 'device': 'CPU'}),
  (<DtypeKind.INT: 0>, 32, 'i', '=')),
 'validity': None,
 'offsets': None}
```
definitely needs fixing

**Comment by MarcoGorelli** [comment]
the bufsize is wrong in the pandas case too, right? the api reads

> Buffer size in bytes.

1 byte is 8 bits, and 32 bits / 8 bits = 4. Not 8

**Comment by WillAyd** [comment]
Good catch - yea that also is wrong

**Comment by MarcoGorelli** [comment]
Fixed by #57764 (thanks for your review there!)

```python
In [3]: >>> df = pd.DataFrame([[None]], dtype="int32[pyarrow]")
   ...: >>> df.__dataframe__().get_column(0).describe_null
Out[3]: (<ColumnNullType.USE_BITMASK: 3>, 0)

In [4]: df.__dataframe__().get_column(0).get_buffers()
Out[4]: 
{'data': (PandasBuffer[pyarrow]({'bufsize': 4, 'ptr': 140157248376896, 'device': 'CPU'}),
  (<DtypeKind.INT: 0>, 32, 'i', '=')),
 'validity': (PandasBuffer[pyarrow]({'bufsize': 1, 'ptr': 140157248376832, 'device': 'CPU'}),
  (<DtypeKind.BOOL: 20>, 1, 'b', '=')),
 'offsets': None}
```
